pkg/git/libgit2/managed/ssh.go

```diff
@@ -117,7 +122,6 @@ type connection struct {
 	session       *ssh.Session
 	currentStream *sshSmartSubtransportStream
 	connected     bool
-	m             sync.RWMutex
```

Comment:
I like this approach, no more locking and safer.
This whole `connection` type was created in order to group these variables and lock them together. Since we no longer lock with this change, we can put them back into `sshSmartSubtransport` struct.

Reply:
Pushed a commit with this change, as Paulo is away at the moment.